File: array/189.py
class Solution:
    def rotate(self, nums, k) :
        count,n=0,len(nums)
        k=k%n
        for i in range(n):
            next,temp=i,nums[i]
            while(True):
                next = (next+k)%n 
                temp,nums[next] = nums[next],temp
                count+=1
                if next==i:
                    break
            if count==n :
                break


def main():
    c=Solution()
    l=[1,2,3,4,5,6,7]
    c.rotate(l,3)
    print(l)

if __name__ =='__main__':
    main()

File: array/189.js
var rotate = function(nums, k) {
  let count=0,n=nums.length;
  k=k%n;
  for(let i=0;i<n;i++){
    let cur=nums[i],next_i=i,temp=0;
    while(true){
      next_i=(next_i+k)%n;
      temp = nums[next_i];
      nums[next_i] = cur;
      cur=temp;
      count++;
      if(next_i==i){
        break;
      }
    }
    if (count==n){
      return
    }
  }
};

let t=[1,2,3,4,5,6];
rotate(t,3);
console.log(t);